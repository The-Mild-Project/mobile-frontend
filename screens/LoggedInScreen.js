import * as React from 'react';
import {Platform, StyleSheet, Text, View} from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';
import { Button } from 'react-native-elements';



/* navigation */
import { createStackNavigator, createAppContainer, NavigationContainer  } from 'react-navigation';
import RestaurantsScreen from "./RestaurantsScreen";

function LoggedInScreen({route, navigation}) {
    let name = route.params.name;
    let email = route.params.email;

    return (
        <ScrollView style={styles.container} contentContainerStyle={styles.contentContainer}>
            <View style={styles.container}>
                <Text>Hi {name}, welcome to the Food App</Text>
                <View style={styles.buttonContainer}>
                    <Button style={styles.smallCenterButton} title="Browse Restaurants" onPress={() => navigation.navigate('Restaurants', {"name": name, "email": email})} />
                    <Button style={styles.smallCenterButton} title="View Recommendations" onPress={() => navigation.navigate('Recs', {"name": name, "email": email})} />
                    <Button style={styles.smallCenterButton} title="Configure Food Preferences" onPress={() => navigation.navigate('Preferences', {"name": name, "email": email})} />
                </View>
            </View>
        </ScrollView>
    );
}

LoggedInScreen.navigationOptions = () => {

};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        paddingLeft: 15,
        paddingRight: 15
    },

    contentContainer: {
        paddingTop: 30,
    },

    smallCenterButton: {
        width: 300,
        height: 40,
        borderRadius: 4,
        marginBottom: 5,
        textAlign: 'center'
    },

    buttonContainer: {
        paddingTop: 15,
        justifyContent: 'center',
        alignItems: 'center'
    }
});

export default LoggedInScreen