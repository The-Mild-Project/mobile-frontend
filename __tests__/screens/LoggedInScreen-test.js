import * as React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import LoggedInScreen from '../../screens/LoggedInScreen';

jest.mock('expo', () => ({
    AppLoading: 'AppLoading',
}));

describe('App', () => {
    jest.useFakeTimers();

    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it(`renders the loading screen`, () => {
        const tree = renderer.create(<LoggedInScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
