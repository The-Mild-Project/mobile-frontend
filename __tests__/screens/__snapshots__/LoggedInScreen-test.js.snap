// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the loading screen 1`] = `
<RCTScrollView
  collapsable={false}
  contentContainerStyle={
    Object {
      "paddingTop": 30,
    }
  }
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingLeft": 15,
      "paddingRight": 15,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    >
      <Text>
        Hi 
        , welcome to the Food App
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 4,
                "height": 40,
                "marginBottom": 5,
                "opacity": 1,
                "textAlign": "center",
                "width": 300,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Browse Restaurants
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 4,
                "height": 40,
                "marginBottom": 5,
                "opacity": 1,
                "textAlign": "center",
                "width": 300,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                View Recommendations
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 4,
                "height": 40,
                "marginBottom": 5,
                "opacity": 1,
                "textAlign": "center",
                "width": 300,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Preferences
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 4,
                "height": 40,
                "marginBottom": 5,
                "opacity": 1,
                "textAlign": "center",
                "width": 300,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                Settings
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
